ui/components/resolvers/ResolverDetail.js

```diff
@@ -55,6 +62,44 @@ class ResolverDetail extends Component {
         this.updateResolver({ responseMapping: template.value });
     }
 
+    onDataSourceSelect(DataSource) {
+        const dsValidation = DataSource !== null ? "success" : "error";
+
+        const { validations } = this.state;
+        const newValidations = { ...validations, dataSource: dsValidation };
+
+        this.setState({ validations: newValidations });
+        this.updateResolver({ DataSource });
+    }
+
+    onPreHookChange(preHook) {
+        const preHookValidation = Validate([
+            Validators.String.minLength(1), preHook,
+            Validators.String.maxLength(255), preHook,
```

Comment:
Could be too small for a URL. The limit seems to be closer to 2000 chars: https://stackoverflow.com/questions/417142/what-is-the-maximum-length-of-a-url-in-different-browsers